docuilib/src/components/ComponentPage.tsx

```diff
@@ -0,0 +1,417 @@
+import _ from 'lodash';
+import React from 'react';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import CodeBlock from '@theme/CodeBlock';
+
+export default function ComponentPage({component}) {
+  const getTitleSize = type => {
+    switch (type) {
+      case 'hero':
+        return 48;
+      case 'item':
+        return 16;
+      default:
+        return 32;
+    }
+  };
+
+  const getTitleWeight = type => {
+    switch (type) {
+      case 'item':
+        return '400';
+      default:
+        return '700';
+    }
+  };
+
+  const getDescriptionColor = type => {
+    switch (type) {
+      case 'item':
+        return '#6E7881';
+      default:
+        return '#495059';
+    }
+  };
+
+  const getTypeColor = (type?: string) => {
+    switch (type) {
+      case 'string':
+        return '#FFEEB9';
+      case 'number':
+        return '#B3EBDD';
+      case 'boolean':
+        return '#C4DFFF';
+      default:
+        return '#E8ECF0';
+    }
+  };
+
+  const getCodeExample = () => {
+    return <a href={component.example} style={{fontSize: '16px', fontWeight: '700', lineHeight: '18px', borderBottom: '1px solid'}}>Code Example</a>;
+  };
+
+  const getTitle = (type, title) => {
+    const size = getTitleSize(type);
+    const weight = getTitleWeight(type);
+    return <span style={{fontSize: size, fontWeight: weight}}>{title}</span>;
+  };
+
+  const getHeader = (title, description, type) => {
+    const desColor = getDescriptionColor(type);
+
+    if (!title && !description) {
+      return;
+    }
+
+    switch (type) {
+      case 'usage':
+        return (
+          <div
+            style={{
+              display: 'flex',
+              flexDirection: 'row',
+              alignItems: 'center',
+              justifyContent: 'space-between',
+              margin: '0 0 12px 0'
```

Comment:
I might have confused you when we talked on margins, while there are no marginHorizontal/Vertical in css you do have independent margin prop for Top/Left/Right/Bottom, you can use them when setting margin on one side only

Reply:
I gave ti to the default (line 85), this is the Usage section's header